skycatalogs/skyCatalogs.py

```diff
@@ -94,48 +109,63 @@ def _compress_via_mask(tbl, id_column, region, galaxy=True):
             dec_min = min([v[1] for v in vertices])
             bnd_box = Box(ra_min, ra_max, dec_min, dec_max)
             # Compute mask for that box
-            mask = _compute_mask(bnd_box, tbl['ra'], tbl['dec'])
+            mask = _compute_region_mask(bnd_box, tbl['ra'], tbl['dec'])
             if all(mask): # even bounding box doesn't intersect table rows
-                if galaxy:
+                if no_obj_type_return:
```

Comment:
There are those source types which don't need object type returned (currently `'galaxy'` and `'snana'`) and those that do  (`'star'` and `'sncosmo'`).  Instead of saying
`if no_obj_type_return:` in all those places the code could say
`if source_type in {'galaxy', 'snana'}:`
everywhere, but the branches are still all necessary.  It seems better to have the evaluation in one place, especially since more source types may be supported with time. 
Longer term, we could consider storing `star` and `sncosmo` in separate files (especially since we'll probably stop using `sncosmo` in new catalogs) and agree to always keep separate source types in separate files. Then everything would fall in the `no_obj_type_return` category.

Reply:
It's not possible to return the object type for galaxy and snana and have the calling code just ignore those entries?  It seems like one could then also simplify the code in places like `get_object_type_by_hp`.  Anyway, this doesn't need to be addressed in this PR.

Reply:
There would still have to be branching since galaxy and snana don't natively have object_type as a column in the parquet file.

Reply:
Let's defer this one.